react/components/EditorContainer/Sidebar/ComponentEditor/utils.ts

```diff
@@ -37,45 +38,32 @@ const getUiSchema = (
    *
    * @param {object} properties The schema properties to be analysed.
    */
-  const getDeepUiSchema = (properties: any): UISchema => {
-    const deepProperties = pickBy(
-      property => has('properties', property),
-      properties
-    )
-    const itemsProperties = pickBy(
-      property => has('properties', property),
-      properties.items
-    )
 
-    return {
-      ...map(
-        (value: ComponentSchema) => value.widget,
-        pickBy(property => has('widget', property), properties)
-      ),
-      ...(deepProperties &&
-        map(property => getDeepUiSchema(property.properties), deepProperties)),
-      ...(itemsProperties &&
-        map(item => getDeepUiSchema(item), itemsProperties)),
+  const arrayAttrs = /(anyOf|oneOf|allOf|dependencies)(\/\d+|\/\w+)?/g
+  const noUISchemaProp = [
+    'definitions',
+    'properties',
+    'patternProperties',
+    'additionalProperties',
+  ]
+  let uiSchema = {}
+
+  const getWidget = (...args: any[]) => {
+    const [schema, JSONPointer] = args
+    if (schema.widget) {
+      const widgetPath = JSONPointer.replace(arrayAttrs, '')
+        .split('/')
+        .filter(
+          (pointer: string) => pointer && !noUISchemaProp.includes(pointer)
+        )
+      uiSchema = {
```

Comment:
Is there any case where a deep merge is necessary? Otherwise, I think the first `...uiSchema` is unnecessary.

Reply:
Well pointed! Tks